Improve background dim API

MortimerGoro opened the pull request:
- Better API to set up the background dim value and go back to the previous ones. The current code breaks a lot.
- Fixes #657
- Provide more flexibility for the configurable brightness in the Video mode.

bluemarvin commented on app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/WidgetManagerDelegate.java:
```diff
@@ -25,8 +27,9 @@
-    void fadeOutWorld();
-    void fadeInWorld();
+    void pushWorldBrightness();
+    void setWorldBrightness(float aBrightness);
```
Do we need separate `setWorldBrightness()` and `pushWorldBrightness()` functions? Are there instances where we want to set the brightness and not have it push the current state?

MortimerGoro replied:
all the cases should call `setWorldBrightness` after `pushWorldBrightness`. So we can add the parameter to setWorldBrightness. I'll update the PR

MortimerGoro commented:
I fixed the mentioned cases:
- I added a key for each brigthness setter/push so if a widget back in the hierarchy changes it's brightness it doesn't affect the current one (e.g. a tab changing from private to normal while a dialog in on top)
- Crash dialog was failing because there is a FadeIn() after the splash is shown. Now that fadeIn uses the correct final brightness.
